Small fixes to flash_backfit, r1_opt

willwerscheid opened the pull request:
Display verbose output for iteration 1 when backfitting; set values for all variables in r1_opt to avoid erroring out.

stephens999 commented on R/r1_opt.R:
```diff
@@ -105,9 +114,8 @@ You could ignore this warning, but you might like to check out https://github.co
-  return(list(l=l,f=f,l2=l2,f2=f2,tau=tau,F_obj=F_obj,KL_l = KL_l, KL_f = KL_f,
-              gl = ebnm_l$fitted_g, gf = ebnm_f$fitted_g,
-              penloglik_l = ebnm_l$penloglik, penloglik_f = ebnm_f$penloglik,
+  return(list(l=l, f=f, l2=l2, f2=f2, tau=tau, F_obj=F_obj, KL_l=KL_l, KL_f=KL_f,
```
don't you need to set 
``gl = ebnm_l$fitted_g, gf = ebnm_f$fitted_g, penloglik_l = ebnm_l$penloglik, penloglik_f = ebnm_f$penloglik``
before this? I guess i don't see where gl etc are being set (except to NULL at initialization)

willwerscheid commented:
Fixed r1_opt variable setting issues. Also fixed some divide by zero problems when calc_F = FALSE (even though we don't use this parameter setting anywhere).
